### ISSUE TITLE
Missing skip link on https://clarity.design

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#3153]
## Description   
Missing skip link on https://clarity.design

## Location      
All pages

## Priority
High
      
## Recommended fix 
https://clarity.design should have a skip link at the  beginning of the DOM to allow sighted keyboard-only users as well assistive technology users the ability to skip past the main navigation to the main content of the page. The skip link should be visually hidden but still available to assistive technologies when it does not have focus. It should become visible when it gains focus and visibly hidden on blur. See an example at https://www.starbucks.com. Another example can be found at http://www.chrislane.info/examples/skipLink.html.
      
## LOE 
Medium

**Comment by Jinnie** [comment]
@coryrylan we have it on the ks-app, I guess we can unify the experience.

**Comment by coryrylan** [comment]
I have an implementation working. The skip link shows when you first tab on the page and then disappears when you tab away as described in the examples above. When selected it will focus to the content of the page. I made it an application wide alert but happy to change it to whatever design recommends. Any thoughts @colinreedmiller ?

![9zirhcywlr](https://user-images.githubusercontent.com/2021067/53974435-3054a000-40c8-11e9-8a70-30ca8df7dba9.gif)

**Comment by colinreedmiller** [comment]
I think the app level alert is the appropriate component. Thoughts @chlane ?

**Comment by gnomeontherun** [comment]
What about when we want to use an app level alert? We have this on the homepage currently.

I think it needs to be a different experience.

**Comment by mathisscott** [comment]
Most of the time a skiplink is its own thing. I agree with @gnomeontherun 

See https://webaim.org/ for a good example.

An app alert may be an expedient choice but I don't think it's the right choice.

**Comment by chlane** [comment]
Hi Colin,

If we are talking about using role=”alert” to notify screen reader users that new content has appeared as  a result of routing I don’t think it is the right fit. Simply moving focus to a heading that precedes the newly loaded content is the best way.

Crhis

**Comment by chlane** [comment]
Hey Jeremy,

If there is a good reason to aggressively interrupt a screen reader user’s reading experience, something like a session timeout warning,  then role=”alert” can be used. When role=”alert” is used, it is not necessary to include aria-live=”assertive”.

Chris

**Comment by coryrylan** [comment]
Agree looking at other examples making skip link its own thing and not an alert. I like @mathisscott suggestion with https://webaim.org/ The skip link on https://www.starbucks.com is kind of funny just a floating button.

@colinreedmiller Maybe take a look at those examples for ideas? We wont use the app level alert component since it conflicts with any existing alerts on the page.

**Comment by Jinnie** [comment]
I already mentioned the kitchensink app. There we added the skip link as "invisible" 1st item on the menu that only gets visible when tabbed to. What's wrong in reusing this approach?
<img width="986" alt="Screen Shot 2019-03-11 at 2 09 55 PM" src="https://user-images.githubusercontent.com/1798828/54123098-864a8000-4407-11e9-8653-b8503e25d8b2.png">

**Comment by Jinnie** [comment]
Sry, I hit wrong "comment" button. Reopening.

**Comment by coryrylan** [comment]
I didn't follow the KS app example for placement mainly because I was hoping to find a solution to the skip link that is independent of the navigation and content of the application. For example, some pages on the clarity site do not have the side nav immediately available. If we can find a generic solution, then we can point users to the docs for an example that they can use in their applications.

**Comment by Jinnie** [comment]
Still, you need to have something to skip :)
If you don't have a menu, what is it that you are skipping? IMO, the idea behind "skip to content" is exactly to skip a possibly lengthy vertical nav. Horizontal ones are not so big, though yes, they can be a candidate for skipping, too. Open for other points of view, of course!

**Comment by chlane** [comment]
Yes there should be something to skip. The idea is to provide a mechanism to skip past repetitive navigation structures. There is no hard and fast rule regarding how many controls are not worth using a skip link. It’s good to simply provide skip links for navigation menus both vertical and horizontal.

Also, the skip link violation is frequently flagged by automated accessibility checkers so it’s good idea just to pass the checks as well.

**Comment by Jinnie** [comment]
So, Cory's initial approach seems better. We'll always have at least the top navigation on the site to skip.

**Comment by colinreedmiller** [comment]
The initial approach was critiqued on the basis of the use case when an app level alert is already present or needed simultaneously to the skiplink. I spoke at length with @chlane about this and have drafted a solution quite similar to what we see in Webaim. I need to check with Sheri re: placement in the visual hierarchy because the layout of https://clarity.design/documentation leaves no obvious location without either changing, occlusion, or placement where it appears lower in the hierarchy than expected.

**Comment by colinreedmiller** [comment]
Design spec.

![skiplink-#3153-design-spec.png](https://images.zenhubusercontent.com/5bc78fe0ba9b001064ad7062/0d0547d3-82ab-4709-8c4d-a19db682b600)

Considerations:
Highly visible is essential - keyboard users often move quickly and are more likely to miss an item that is invisible - then visible - then invisible again, thus over afforded (color) is good, especially in a tight layout that should not dynamically change with the appearance of the skiplink.

also: Using this opportunity to align the header tabs with the side nav (see spec).

Below - appearance on other Clarity tabs with and without side nav:

![Community.png](https://images.zenhubusercontent.com/5bc78fe0ba9b001064ad7062/ffe92608-3e1d-44c2-bda9-6c9b72b8cf8e)
![Icons.png](https://images.zenhubusercontent.com/5bc78fe0ba9b001064ad7062/8b9dd3a7-5101-47d3-9823-0ae2581ee1dc)
![Icons.png](https://images.zenhubusercontent.com/5bc78fe0ba9b001064ad7062/de516836-09e2-4d11-aed9-549de3a9585d)
